### PR TITLE
Feature: offline first setup

### PR DESCRIPTION
**Pull request by sorodrigo**
This PR includes the following:
- Redux offline setup
- Removes connection status listener (uses redux offline instead)
- Refactors google sign in into a singleton object (also fixes logout issues).
- Custom redux-offline effect implementation.

Although the majority of changes needed to use redux-offline are still missing, I think its best to limit the size of the PR to what we have now. More PRs related to offline incoming, hold on tight.

### CODE REVIEW

**Comment by j8seangel** [comment]
The old react-native-debugger is not working anymore as the offline compose didn't work well with the composeWithDevTools so I gave a chance to [reactotron](https://github.com/infinitered/reactotron/) that also give us more information as the network request or the async storage usage, so just install and open it to have it available!

File: app/components/login/index.js [hunk 1]
```diff
@@ -9,12 +9,12 @@ import {
   Image
 } from 'react-native';
 
-import { GoogleSignin } from 'react-native-google-signin';
 import Config from 'react-native-config';
 import Theme from 'config/theme';
 import I18n from 'locales';
 import tracker from 'helpers/googleAnalytics';
 
+import GoogleOAuth from '../../config/oAuth/GoogleOAuth';
```
Comment (j8seangel) [inline app/components/login/index.js]: As we are using alias we don't need the relative path.

File: app/main.js [hunk 2]
```diff
@@ -1,56 +1,55 @@
...
 import thunk from 'redux-thunk';
 import { Provider } from 'react-redux';
-import { persistStore, autoRehydrate } from 'redux-persist';
-import { composeWithDevTools } from 'redux-devtools-extension';
 
 import Theme from 'config/theme';
 import { registerScreens } from 'screens';
 
 import * as reducers from 'redux-modules';
+import offline from './offline';
```
Comment (j8seangel) [inline app/main.js]: Could you add offline as alias here https://github.com/Vizzuality/forest-watcher/blob/feature/dropdown-for-viirs-time/package.json#L73 ? 
Thanks !

Comment (j8seangel) [inline app/main.js]: And this change is awesome 😍 !

File: app/redux-modules/user.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import Config from 'react-native-config';
-import { GoogleSignin } from 'react-native-google-signin';
+import GoogleOAuth from '../config/oAuth/GoogleOAuth';
```
Comment (j8seangel) [inline app/redux-modules/user.js]: Please use that alias here.

File: app/redux-modules/user.js [hunk 4]
```diff
@@ -114,9 +115,14 @@ export function setLoginStatus(status) {
 
 export function logout() {
   return async (dispatch) => {
-    await GoogleSignin.revokeAccess();
     dispatch({
-      type: LOGOUT
+      type: LOGOUT_ATTEMPT,
```
Comment (j8seangel) [inline app/redux-modules/user.js]: What do you think about using a standard way for the actions? 
For example, as they do in the redux-offline library:
- LOGOUT_REQUEST
- LOGOUT_COMMIT
- LOGOUT_ROLLBACK (we are very optimistic and this never happens haha)

File: app/main.js [hunk 5]
```diff
@@ -1,56 +1,55 @@
...
+      screen: {
+        screen: 'ForestWatcher.Home',
+        navigatorStyle: Theme.navigator.styles
+      }
     });
   }
-}
 
-export default App;
+  const authMiddleware = ({ getState }) => next => action => next({ ...action, auth: getState().user.token });
```
Comment (j8seangel) [inline app/main.js]: Our first customMidleware is so useful 👏 👏

**Comment by sorodrigo** [comment]
Ready 👍

**Comment by j8seangel** [comment]
Awesome work!

**Pull request merged**